frontend/src/components/ConstraintForm/SubjectAccordionItem.vue

```diff
@@ -0,0 +1,78 @@
+<script>
+import { AccordionItem } from 'vue3-rich-accordion';
+import { useFormStore } from '@/stores/form';
+
+export default {
+  props: {
+    index: { type: Number, default: -1 },
+    classIndex: { type: Number, default: -1 },
+  },
+  setup(props) {
+    const formStore = useFormStore();
+
+    const teachers = ['Ante', 'Ana', 'Mate']; // TODO: replace with real data when teacher crud frontend is finished
+    const classrooms = ['001', '002', '003']; // TODO: replace with real data when classroom crud frontend is finished
+
+    const accordionId = `${props.classIndex}_${props.index}`;
+
+    return { formStore, teachers, classrooms, accordionId };
+  },
+  components: {
+    AccordionItem,
+  },
+};
+</script>
+
+<template>
+  <accordion-item
+    :id="accordionId"
+    :default-opened="accordionId in formStore.accordionState ? formStore.accordionState[accordionId] : true"
+    class="subject-accordion"
+  >
+    <template #summary>
+      <input
+        v-model="formStore.form.classes[classIndex].subjects[index].name"
```

Comment:
Similar to the comment above, `formStore.form.classes[classIndex].subjects[index]` can be placed in a variable in the `setup` method.